src/wish.cpp

```diff
@@ -476,36 +485,42 @@ void debug_menu::wishitem( player *p, int x, int y, int z )
         wmenu.query();
         if( wmenu.ret >= 0 ) {
             item granted( opts[wmenu.ret] );
+            if( dynamic_cast<wish_item_callback *>( wmenu.callback )->incontainer ) {
```

Comment:
Why don't you use the callback object `cb` directly? That way you won't need the `dynamic_cast`.